packages/amplify_auth_cognito/ios/Classes/FlutterSignInWithWebUIRequest.swift

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import Foundation
+import Amplify
+import AmplifyPlugins
+
+struct FlutterSignInWithWebUIRequest {
+  var provider: AuthProvider?
+
+  init(dict: NSMutableDictionary){
+    self.provider = getAuthProvider(provider: dict["authProvider"] as! String?)
+  }
+
+  func getAuthProvider(provider: String?) -> AuthProvider? {
+    if (provider != nil) {
+      switch provider {
+        case "facebook":
+          return AuthProvider.facebook
+        case "login_with_amazon":
+          return AuthProvider.amazon
+        case "google":
+          return AuthProvider.google
+        default:
+          return nil
+      }
+    }
+    return nil
+  }
```

Comment:
Isn't provider is null for `SignInWithWebUI`

Reply:
The request itself may or may not have this value.  In iOS, we use the presence of this value to determine whether or not to call signInWithWebUI with a `for` parameter.

Reply:
The request itself may or may not have a provider. This is used to determine which method to call on the bridge (one of the methods uses a `for` parameter when calling the amplify-ios `signInWithWebUI`method, the other does not).

Reply:
But don't you already have two classes to represent these requests? This and `FlutterSocialSignInWithWebUIRequest`. It doesn't look like you are using `FlutterSocialSignInWithWebUIRequest` class though.